components/ChainSelector.tsx

```diff
@@ -46,18 +47,23 @@ const ChainSelector = () => {
       onForkChange(option.value)
 
       router.query.fork = option.value
-      router.push(router)
+      router.push({ query: router.query })
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [onForkChange],
   )
 
-  useEffect(() => {
-    if (defaultForkOption) {
-      handleForkChange(defaultForkOption)
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [defaultForkOption])
+  // TODO: don't redirect (or at least don't override address query param)
```

Comment:
yes, as I mentioned last time I did not intend to commit this.
this is a temp fix because this part breaks my own router. it keeps redirecting and removing my parameters.
instead of touching your code right now I temporarily disabled this (only on load, the chain selector still works)

eventually we will need to fix it there too.

Reply:
so since this doesn't break anything, it only fixes things, I will mark it as resolved.
we can discuss this chunk of code on another occasion.